lib/flagsmith.rb

```diff
@@ -178,34 +203,65 @@ def get_identity_flags_from_document(identifier, traits = {})
       Flagsmith::Flags::Collection.from_feature_state_models(
         engine.get_identity_feature_states(environment, identity_model),
         analytics_processor: analytics_processor,
-        default_flag_handler: default_flag_handler
+        default_flag_handler: default_flag_handler,
+        offline_handler: offline_handler
       )
     end
 
+    # rubocop:disable Metrics/MethodLength
     def environment_flags_from_api
-      rescue_with_default_handler do
-        api_flags = api_client.get(@config.environment_flags_url).body
-        api_flags = api_flags.select { |flag| flag[:feature_segment].nil? }
-        Flagsmith::Flags::Collection.from_api(
-          api_flags,
-          analytics_processor: analytics_processor,
-          default_flag_handler: default_flag_handler
-        )
+      if offline_handler
+        begin
+          process_environment_flags_from_api
+        rescue StandardError
+          environment_flags_from_document
+        end
+      else
+        rescue_with_default_handler do
```

Comment:
Would this be neater if we actually didn't use this helper method now that we're rescuing in 2 separate ways?

Reply:
Well I certainly didn't love that `rescue_with_default_handler` method, but I didn't hate it enough to go out of my way to refactor it for the PR. But if you're thinking it should just be a plain old `begin` `rescue` lines of code I'll get rid of it and just work the logic into the the method.

Reply:
Ok this has been fixed.